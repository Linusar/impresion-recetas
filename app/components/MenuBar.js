import React from 'react';
import clsx from 'clsx';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import HomeIcon from '@material-ui/icons/Home';
import SaveIcon from '@material-ui/icons/AddBox';
import BuildIcon from '@material-ui/icons/Menu';
import routes from '../constants/routes';
import RedirectLink from './RedirectLink';

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1
  },
  leftIcon: {
    marginRight: theme.spacing(1)
  },
  iconSmall: {
    fontSize: 20
  },
  title: {
    flexGrow: 1
  }
}));

const MenuBar = () => {
  const classes = useStyles();

  return (
    <AppBar position="relative">
      <Toolbar>
        <Typography
          variant="h6"
          className={classes.title}
          data-test="menuBarTypography"
        >
          Impresión de Recetas
        </Typography>
        <Button
          color="inherit"
          component={RedirectLink(routes.HOME)}
          data-test="menuBarButtonHome"
        >
          <HomeIcon
            className={clsx(classes.leftIcon, classes.iconSmall)}
            data-test="menuBarIconHome"
          />
          Home
        </Button>
        <Button
          color="inherit"
          component={RedirectLink(routes.CARGAR)}
          data-test="menuBarButtonCargar"
        >
          <SaveIcon
            className={clsx(classes.leftIcon, classes.iconSmall)}
            data-test="menuBarIconCargar"
          />
          Cargar
        </Button>
        <Button
          color="inherit"
          component={RedirectLink(routes.CONFIG)}
          data-test="menuBarButtonConfig"
        >
          <BuildIcon
            className={clsx(classes.leftIcon, classes.iconSmall)}
            data-test="menuBarIconConfig"
          />
          Configuración
        </Button>
      </Toolbar>
    </AppBar>
  );
};

export default MenuBar;
