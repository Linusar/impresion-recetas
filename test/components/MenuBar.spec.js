import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import HomeIcon from '@material-ui/icons/Home';
import WhatshotIcon from '@material-ui/icons/Whatshot';
import GrainIcon from '@material-ui/icons/Grain';
import MenuBar from '../../app/components/MenuBar';

Enzyme.configure({ adapter: new Adapter() });

const setup = () => {
  const component = shallow(<MenuBar />);
  return {
    component,
    buttons: component.find(Button)
  };
};

describe('Menu Bar render', () => {
  it('should render AppBar', () => {
    const { component } = setup();
    expect(component.find(AppBar).exists()).toBe(true);
  });

  it('should render Toolbar', () => {
    const { component } = setup();
    expect(component.find(Toolbar).exists()).toBe(true);
  });

  it('should display text "Impresión de Recetas"', () => {
    const { component } = setup();
    expect(component.find(Typography).text()).toBe('Impresión de Recetas');
  });

  it('should display Home button', () => {
    const { buttons } = setup();
    buttons.at(0).simulate('click');
    expect(buttons.at(0).exists()).toBe(true);
  });

  it('should display Cargar Recetas button', () => {
    const { buttons } = setup();
    buttons.at(1).simulate('click');
    expect(buttons.at(1).exists()).toBe(true);
  });

  it('should display Configuración button', () => {
    const { buttons } = setup();
    buttons.at(2).simulate('click');
    expect(buttons.at(2).exists()).toBe(true);
  });

  it('should display HomeIcon', () => {
    const { component } = setup();
    expect(component.find(HomeIcon).exists()).toBe(true);
  });

  it('should display WhatshotIcon', () => {
    const { component } = setup();
    expect(component.find(WhatshotIcon).exists()).toBe(true);
  });

  it('should display GrainIcon', () => {
    const { component } = setup();
    expect(component.find(GrainIcon).exists()).toBe(true);
  });

  it('should match exact snapshot', () => {
    const menu = (
      <div>
        <Router>
          <MenuBar />
        </Router>
      </div>
    );
    const tree = renderer.create(menu).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
