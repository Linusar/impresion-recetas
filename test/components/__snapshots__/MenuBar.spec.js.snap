// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Bar render should match exact snapshot 1`] = `
<div>
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root makeStyles-title-4 MuiTypography-h6"
        data-test="menuBarTypography"
      >
        Impresión de Recetas
      </h6>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        data-test="menuBarButtonHome"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-leftIcon-2 makeStyles-iconSmall-3"
            data-test="menuBarIconHome"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
          Home
        </span>
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        data-test="menuBarButtonCargar"
        href="/cargar"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-leftIcon-2 makeStyles-iconSmall-3"
            data-test="menuBarIconCargar"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
          Cargar
        </span>
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        data-test="menuBarButtonConfig"
        href="/config"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-leftIcon-2 makeStyles-iconSmall-3"
            data-test="menuBarIconConfig"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          Configuración
        </span>
      </a>
    </div>
  </header>
</div>
`;
